store/src/java/com/zimbra/cs/mailbox/Folder.java

```diff
@@ -806,7 +812,11 @@ boolean canContain(MailItem child) {
             // may not contain our parents or grandparents (c.f. Back to the Future)
             for (Folder folder = this; folder.getId() != Mailbox.ID_FOLDER_ROOT; folder = folder.parent) {
                 if (folder.getId() == child.getId()) {
-                    return false;
+                        return false;
+                }
+                if (folder.parent == null) {
+                    folder.setupParent();
```

Comment:
@mamtadahiwal It will always return false even if the parent is set in this method. You need to double check for null here

Reply:
if (folder.parent == null) {
                    folder.setupParent();
                    if (folder.parent == null) {
                        return false;
                    }
                } like this ur saying